internal/http/http.go

```diff
@@ -264,7 +293,7 @@ func newUploadRequest(target, username, secret string, headers map[string]string
 	return req, err
 }
 
-func getHTTPClient(put *config.Put) (*h.Client, error) {
+func getHTTPClient(put *config.Upload) (*h.Client, error) {
```

Comment:
Since we're renaming `put` variables as `upload` everywhere, this probably needs to renamed too.